src/null-safety/faq.md

```diff
@@ -301,6 +301,77 @@ and situations where a null value is really expected. So the tool tells you what
 it knows ("it looks like this condition will always be false!") and lets you
 decide what to do.
 
+## What should I know about dart2js and null safety?
+
+For a long time, the dart2js compiler has had optimizations specifically
+targeting null values and null checks. Because of that, null safety is not
+expected to affect much the output of the compiler.
+
+A few notes that are worth highlighting:
+
+* dart2js emits `!` null assertions, but you may not notice them.
+  That's because pre null-safety dart2js already emitted null checks
+  (so they are already in the existing programs).
+
+* dart2js emits these null assertions regardless of sound/unsound null safety
```

Comment:
Based on questions I've seen around this exact output, I can see this section causing more questions about "why" this is the case. Is there anything we can do to more directly connect to or elaborate the "why"?

Reply:
Thanks! Yeah I think even a small clarification around maintaining program semantics within this point would be sufficient, as otherwise it may not be immediately clear to the average reader.